packages/drivers/odsp-driver/src/epochTracker.ts

```diff
@@ -153,9 +157,20 @@ export class EpochTracker {
         }
     }
 
-    private async checkForEpochError(error) {
+    private async checkForEpochError(
+        error: any,
+        epochFromResponse: string | null | undefined,
+        fromCache: boolean = false,
+    ) {
         if (error.errorType === OdspErrorType.epochVersionMismatch) {
-            this.logger.sendErrorEvent({ eventName: "EpochVersionMismatch" }, error);
+            this.logger.sendErrorEvent(
+                {
+                    eventName: "EpochVersionMismatch",
+                    fromCache,
+                    clientEpoch: this.fluidEpoch,
+                    epochFromResponse: epochFromResponse ?? undefined,
```

Comment:
maybe serverEpoch (for consistency with clientEpoch)?

Reply:
I thought of that but it can from cache too. So I used epochFromResponse instead of serverResponse.

Reply:
sorry, but "from cache" can apply to clientEpoch, not to serverEpoch, right?
So I think we should use client & server terminology, and know that client info may have come from cache or from server earlier

Reply:
In that case, I think we should use oldEpoch/newEpoch because lets say epoch tracker has epoch "x" and then we fetch something from cache, then epoch coming from cache would be "y", in that case there is no server epoch. That is why I used epochFromResponse because response could come from cache or server.

Reply:
And "fromCache" tells whether the response was from cache or server.

Reply:
sure. Though fromCache is for oldEpoch, not for newEpoch, right?
We do not have have yet scenarios where data is coming from cache other than snapshot, so any mismatch is due to data coming from server.

Reply:
yeah in that sense, I can rename it too client/serverEpoch.